src/hashtree.h

```diff
@@ -23,18 +23,32 @@ SOFTWARE.
 */
 #ifndef HASHTREE_H
 #define HASHTREE_H
+
 #include <stdint.h>
 
-void hashtree_init();
+typedef void (*hashtree_hash_fcn)(unsigned char*, const unsigned char*, uint64_t);
+
+/** Perfrom hardware detection and return the fastest function available
+ *
+ * @returns Fastest hash tree function, or NULL if no supported hardware is found
+ */
+hashtree_hash_fcn hashtree_detect();
+
+/** Initialize the library to use the given hash tree function or perform
+ * auto-detection based on the CPU if `NULL` is given.
+ *
+ * @returns 0 if auto-detection failed, non-0 otherwise
```

Comment:
I would prefer typically 0 to mean no failures.

Reply:
this is the `bool` style which fits neatly in an `if(hashtree_init..)` - if we go with 0, I think we need a `HASHTREE_OK` constant to compare against, wdyt?